### PR TITLE
ARROW-4127: [Documentation] Add docker documentation instructions

### PR DESCRIPTION
**Pull request by cav71**
adds docker documentation instructions
(https://issues.apache.org/jira/browse/ARROW-4127)

### CODE REVIEW

File: docs/source/python/development.rst [hunk 1]
```diff
@@ -414,3 +416,20 @@ After these steps are completed, the documentation is rendered in HTML
...
 at ``docs/_build/html/index.html`` to read the docs and review any changes
 you made.
+
+
+Building with Docker
+--------------------
+
+You can use Docker to build the documentation:
+
+   .. code-block:: shell
```
Comment (kou) [inline docs/source/python/development.rst]: It seems that this indentation isn't needed.

File: docs/source/python/development.rst [hunk 2]
```diff
@@ -414,3 +416,20 @@ After these steps are completed, the documentation is rendered in HTML
...
+
+        docker-compose build cpp 
+        docker-compose build python 
+        docker-compose run docs
```
Comment (kou) [inline docs/source/python/development.rst]: There are too much indentations.

**Review by kou — CHANGES_REQUESTED** [review]
Thanks.
Can you remove changes for `cpp/submodules/parquet-testing`?

**Comment by cav71** [comment]
Opened a new pull request https://github.com/apache/arrow/pull/3281

File: docs/source/python/development.rst (hunk 2 above)
Comment (cav71) [reply docs/source/python/development.rst]: Fixed in the new pull request https://github.com/apache/arrow/pull/3281 thanks

**Pull request closed without merging**